### PR TITLE
chameleon_hier updates for BLOCKS

### LINKED ISSUE #3322: Error: global_route.tcl, 21 GRT-0119
### Describe the bug

When I try to run make DESIGN_CONFIG=./designs/sky130hd/chameleon_hier/config.mk, I am getting an error. 

[INFO GRT-0101] Running extra iterations to remove overflow.
[WARNING GRT-0230] Congestion iterations cannot increase overflow, reached the maximum number of times the total overflow can be increased.
[INFO GRT-0197] Via related to pin nodes: 14773
[INFO GRT-0198] Via related Steiner nodes: 555
[INFO GRT-0199] Via filling finished.
[INFO GRT-0111] Final number of vias: 22592
[INFO GRT-0112] Final usage 3D: 321877
[ERROR GRT-0119] Routing congestion too high. Check the congestion heatmap in the GUI and load ./reports/sky130hd/chameleon_hier/base/congestion.rpt in the DRC viewer.
Error: global_route.tcl, 21 GRT-0119
Command exited with non-zero status 1
Elapsed time: 0:33.32[h:]min:sec. CPU time: user 32.20 sys 1.10 (99%). Peak memory: 2623396KB.
make: *** [Makefile:598: results/sky130hd/chameleon_hier/base/5_1_grt.odb] Error 1

### Expected Behavior

A clean RTL-to-GDS flow

### Environment

```shell
n/a
```

### To Reproduce

n/a

### Relevant log output

_No response_

### Screenshots

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by vijayank88**
`MIN/MAX_ROUTING_LAYER` updated as user configurable in `sky130hd/config.mk`

`chameleon_hier` routing congestion issue fixed and resolves https://github.com/The-OpenROAD-Project/OpenROAD/issues/3322 

Still PSM failing during finish for `ibex_wrapper`. So skipped as of now IR drop check.

### CODE REVIEW

**Comment by vijayank88** [comment]
Metrics fail for `nangate45/bp_multi_top`
```
[ERROR] finish__timing__drv__max_slew_limit fail test: -0.6368 >= -0.51
```

File: flow/designs/sky130hd/chameleon_hier/config.mk
Comment (maliberty) [inline flow/designs/sky130hd/chameleon_hier/config.mk:46]: Why is such a large halo needed?

Comment (vijayank88) [reply flow/designs/sky130hd/chameleon_hier/config.mk:46]: Less than that causing horizontal and vertical congestion for for apb_xxx_inst0 alone.

**Comment by vijayank88** [comment]
> Metrics fail for `nangate45/bp_multi_top`
> 
> ```
> [ERROR] finish__timing__drv__max_slew_limit fail test: -0.6368 >= -0.51
> ```

@maliberty Shall I update this metrics along with this PR? Even @oharboe facing similar issue with `bp_multi_top`

**Comment by maliberty** [comment]
> @maliberty Shall I update this metrics along with this PR? Even @oharboe facing similar issue with `bp_multi_top`

No I'm updating the OR submodule.  The head of master doesn't have this issue.  You'll have to merge master to your branch after #1068 is merged

File: flow/designs/sky130hd/chameleon_hier/config.mk
Comment (vijayank88) [reply flow/designs/sky130hd/chameleon_hier/config.mk:46]: If I use halo 120 and above no congestion occurs. So I kept halo and channel the same. @maliberty

**Comment by vijayank88** [comment]
@maliberty 
Don't merge this. I'm re-working on `PDN_CFG` to `PDN_TCL`.

**Comment by maliberty** [comment]
> @maliberty Don't merge this. I'm re-working on `PDN_CFG` to `PDN_TCL`.

I'll hold on reviewing until you say you are done.

**Comment by vijayank88** [comment]
@maliberty 
it's ready for review.
Converted `PDN_CFG` to `PDN_TCL` format.
Resolved congestion issue. But `chameleon_hier` has no violations with previous OpenROAD commit. With latest commit facing 9 drc violation after detail routing.
I am trying to reproduce the same. It may take some more times seems.
We will merge this and work on incremental PR for the drc violations fix.

**Review by maliberty — APPROVED** [review]

**Comment by maliberty** [comment]
How long does this design take to run?

**Comment by vijayank88** [comment]
> How long does this design take to run?

Around 2Hrs.

**Comment by maliberty** [comment]
How much of that is in the blocks vs the top level?

**Comment by vijayank88** [comment]
> How much of that is in the blocks vs the top level?

blocks level took 1Hr and top level took 1.40Hrs. Total run time is 2.40Hrs

**Pull request merged**
